src/main/java/org/openhab/tools/analysis/checkstyle/EshInfXmlCheck.java

```diff
@@ -0,0 +1,348 @@
+
+        NodeList nodes = null;
+        try {
+            nodes = (NodeList) xpathExpression.evaluate(document, XPathConstants.NODESET);
+        } catch (XPathExpressionException e) {
+            logger.error("Problem occurred while evaluating the expression {} on the {} file.", expression,
+                    xmlFile.getName(), e);
+        }
+        return nodes;
+    }
+
+    private void validateXmlAgainstSchema(File xmlFile, String schemaPath) {
+        URL schemaURL = getSchemaURL(schemaPath);
+        if (schemaURL != null) {
+            try {
+                SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+                Schema schema = schemaFactory.newSchema(schemaURL);
+
+                Validator validator = schema.newValidator();
+                validator.validate(new StreamSource(xmlFile));
+
+            } catch (SAXParseException exception) {
+                String message = exception.getMessage();
+                // Removing the type of the logged message (For example - "cvc-complex-type.2.4.b: ...").
+                message = message.substring(message.indexOf(":") + 2);
+                int lineNumber = exception.getLineNumber();
+                log(lineNumber, message, xmlFile.getPath());
+            } catch (IOException | SAXException e) {
+                logger.error("Problem occurred while parsing the file " + xmlFile.getName(), e);
+            }
+        } else {
+            logger.warn("Unable to reach {}. XML validation will be skipped.", schemaPath);
+        }
+    }
+
+    private URL getSchemaURL(String schemaPath) {
+        HttpURLConnection connection = null;
+        try {
+            URL url = new URL(schemaPath);
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("HEAD");
+            int responseCode = connection.getResponseCode();
+            if (responseCode != 200) {
+                return null;
+            }
+            return url;
+        } catch (IOException e) {
+            return null;
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+    }
+
+    private <K, V> Map<K, V> intersect(Map<K, V> firstMap, Map<K, V> secondMap) {
+        Map<K, V> result = new HashMap<>(firstMap);
+        result.keySet().removeAll(secondMap.keySet());
+        return result;
+    }
+
+    private <K> void logMissingEntries(Map<K, File> collection, String message) {
+        for (K element : collection.keySet()) {
+            File xmlFile = collection.get(element);
+            logMessage(xmlFile.getPath(), 0, xmlFile.getName(), MessageFormat.format(message, element));
+        }
+    }
+
+    private void logMessage(String filePath, int line, String fileName, String message) {
+        MessageDispatcher dispatcher = getMessageDispatcher();
+        dispatcher.fireFileStarted(filePath);
```

Comment:
Maybe yes, it is used to instead of log(..) in the finishProcessing() method, I think it's definitely worth to add it in the abstract class and document it.